.tekton/scripts/determine-if-e2e-execution-needed.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import json
+import urllib.error
+import urllib.request
+import subprocess
+
+# Pipelines currently covered by e2e tests
+pipelines_covered_by_e2e = ["docker-build", "docker-build-oci-ta", "docker-build-multi-platform-oci-ta", "fbc-builder"]
+
+# Task list which are covered by e2e tests, generated dynamically 
+tasks_covered_by_e2e = []
+
+# Otherthan tasks and pipelines, related files for which e2e tests needs to be executed
+files_covered_by_e2e = [".tekton/pull-request.yaml", ".tekton/tasks/e2e-test.yaml", ".tekton/scripts/determine-if-e2e-execution-needed.py"]
+
+def add_only_unique_task_names(task_list):
+    for task_name in task_list:
+        if task_name not in tasks_covered_by_e2e:
+            tasks_covered_by_e2e.append(task_name)
+    
+def get_tasks_covered_by_e2e():
+    for pipeline_name in pipelines_covered_by_e2e:
+        pipeline_path = f"pipelines/{pipeline_name}/{pipeline_name}.yaml"
+        # Get the task names from pipeline spec
+        result = subprocess.run([f"yq -e '.spec.tasks[].taskRef.name' {pipeline_path}"], shell=True, capture_output=True, text=True)
```

Comment:
```suggestion
        result = subprocess.run(["yq", "-e", ".spec.tasks[].taskRef.name", pipeline_path], capture_output=True, text=True)
```
No need for `shell=True` here. That's generally unsafe (prone to shell injection if one doesn't properly escape variables like the `pipeline_path` here)